Assets/VRTK/Source/Scripts/Utilities/VRTK_SharedMethods.cs

```diff
@@ -261,66 +261,46 @@ public static float Mod(float a, float b)
         {
             if (string.IsNullOrEmpty(gameObjectName))
             {
-                T foundComponent = FindEvenInactiveComponent<T>();
+                T foundComponent = FindEvenInactiveComponentsInLoadedScenes<T>(true).FirstOrDefault();
                 return foundComponent == null ? null : foundComponent.gameObject;
             }
 
-            Scene activeScene = SceneManager.GetActiveScene();
-            IEnumerable<GameObject> gameObjects = Resources.FindObjectsOfTypeAll<T>()
-                                                           .Select(component => component.gameObject)
-                                                           .Where(gameObject => gameObject.scene == activeScene);
-
-#if UNITY_EDITOR
-            gameObjects = gameObjects.Where(gameObject => !AssetDatabase.Contains(gameObject));
-#endif
-
-            return gameObjects.Select(gameObject =>
-                              {
-                                  Transform transform = gameObject.transform.Find(gameObjectName);
-                                  return transform == null ? null : transform.gameObject;
-                              })
-                              .FirstOrDefault(gameObject => gameObject != null);
+            return FindEvenInactiveComponentsInLoadedScenes<T>()
+                       .Select(component => 
+                       {
+                            Transform transform = component.gameObject.transform.Find(gameObjectName);
```

Comment:
I've avoided the extra select on https://github.com/thestonefox/VRTK/pull/1725/files#diff-5fe0a238b22c48e3e2f459d8f78fbe2cL270 by simply referencing the `component.gameObject` here instead.